plugins/repository-azure/src/main/java/org/elasticsearch/repositories/azure/AzureStorageSettings.java

```diff
@@ -140,15 +153,33 @@ public Proxy getProxy() {
         return proxy;
     }
 
+    public String getConnectionString() {
```

Comment:
I have renamed it to `buildConnectionString`.

I slightly prefer to keep `AzureStorageSettings` with only primal members, no derivated/computed ones. It's only a taste decision, because if the connection string would have been build several times, or there would be other such fields, then I would go for it and taint the struct with cached members 😛